scipy/optimize/_optimize.py

```diff
@@ -2994,9 +3014,59 @@ def bracket(func, xa=0.0, xb=1.0, args=(), grow_limit=110.0, maxiter=1000):
         fa = fb
         fb = fc
         fc = fw
+
+    # three conditions for a valid bracket
+    cond1 = (fb < fc and fb <= fa) or (fb < fa and fb <= fc)
+    cond2 = (xa < xb < xc or xc < xb < xa)
+    cond3 = np.isfinite(xa) and np.isfinite(xb) and np.isfinite(xc)
+    msg = ("The algorithm terminated without finding a valid bracket. "
+           "Consider trying different initial points.")
+    if not cond1 and cond2 and cond3:
```

Comment:
Should this not be?
```suggestion
    if not (cond1 and cond2 and cond3):
```

Reply:
Good call. I'll add a more thorough test that would have caught this.